libraries/core_libs/consensus/src/rewards/block_stats.cpp

```diff
@@ -6,33 +6,40 @@
 
 namespace taraxa::rewards {
 
-BlockStats::BlockStats(const PeriodData& block, uint64_t dpos_vote_count, uint32_t committee_size)
+BlockStats::BlockStats(const PeriodData& block, const std::vector<gas_t>& trxs_gas_used, uint64_t dpos_vote_count,
+                       uint32_t committee_size)
     : block_author_(block.pbft_blk->getBeneficiary()),
       max_votes_weight_(std::min<uint64_t>(committee_size, dpos_vote_count)) {
+  initFeeByTrxHash(block.transactions, trxs_gas_used);
   processStats(block);
 }
 
-bool BlockStats::addTransaction(const trx_hash_t& tx_hash, const addr_t& validator) {
-  auto found_tx = validator_by_tx_hash_.find(tx_hash);
+void BlockStats::initFeeByTrxHash(const SharedTransactions& transactions, const std::vector<gas_t>& trxs_gas_used) {
+  // assert(transactions.size() == trxs_gas_used.size());
 
-  // Already processed tx
-  if (found_tx != validator_by_tx_hash_.end()) {
-    return false;
+  for (uint32_t i = 0; i < transactions.size(); ++i) {
+    const auto& trx = transactions[i];
+    uint64_t gas_used = 0;
+    // if its not empty then we will add this to validator stats and distribute it
+    if (!trxs_gas_used.empty()) {
+      gas_used = trxs_gas_used[i];
+    }
+    fee_by_trx_hash_.emplace(trx->getHash(), trx->getGasPrice() * gas_used);
   }
-
-  // New tx
-  validator_by_tx_hash_[tx_hash] = validator;
-
-  return true;
 }
 
-std::optional<addr_t> BlockStats::getTransactionValidator(const trx_hash_t& tx_hash) {
-  auto found_tx = validator_by_tx_hash_.find(tx_hash);
-  if (found_tx == validator_by_tx_hash_.end()) {
-    return {};
+bool BlockStats::addTransaction(const trx_hash_t& trx_hash, const addr_t& validator) {
+  auto itr = fee_by_trx_hash_.find(trx_hash);
+  if (itr == fee_by_trx_hash_.end()) {
+    // No fee record found for transaction means that transaction was processed before
+    return false;
   }
 
-  return {found_tx->second};
+  validators_stats_[validator].fees_rewards_ += itr->second;
```

Comment:
@kstdl I saw in evm that we no longer use tx fees in DistributeRewards function, what happens with txs fees now ?

Reply:
https://github.com/Taraxa-project/taraxa-evm/pull/184/files#diff-926216bbe8192eb41a788c8d0b3f6a4a37932ac63db9613e1ba354e5cfb787c9R705-R713